src/Microphysics2M.jl

```diff
@@ -159,36 +150,35 @@ function pdf_rain_parameters(
 end
 
 """
-    size_distribution(pdf, D, q, ρ, N)
+    size_distribution(pdf, q, ρₐ, N)
+    size_distribution(pdf, q, ρₐ, N, D)
 
- - pdf - struct containing size distribution parameters of cloud or rain
- - D - particle size (i.e. maximum dimension of particle)
- - q - cloud or rain water specific content
- - ρₐ - density of air
- - N - cloud or rain water number concentration
+Calculate the size distribution value for rain or cloud particles for a given particle size.
 
-    Returns the size distribution value for rain or cloud particles for a given
-    particle size
+# Arguments
+- `pdf`: Struct containing size distribution parameters of cloud or rain
+- `q`: Cloud or rain water specific content [kg/kg]
+- `ρₐ`: Density of air [kg/m³]
+- `N`: Cloud or rain water number concentration [1/m³]
+- `D`: Particle size (i.e. maximum dimension of particle) [m]
+
+# Returns
+- The first method returns a function that computes the size distribution value
+    for rain or cloud particles for a given particle size. 
+- The second method returns the size distribution value for rain or cloud particles 
+    for a given particle size.
 """
-function size_distribution(
-    pdf::Union{
-        CMP.RainParticlePDF_SB2006{FT},
-        CMP.RainParticlePDF_SB2006_limited{FT},
-    },
-    D::FT,
-    q::FT,
-    ρₐ::FT,
-    N::FT,
-) where {FT}
+function size_distribution(pdf::CMP.RainParticlePDF_SB2006_MaybeLimited, q, ρₐ, N, D)
```

Comment:
I don't think that maybe limited is a good name. We can just pass here a `Union` of the two SB PDF options.